Create notebook introducing workspace resources

emmarogge opened the pull request:
Create notebook introducing the types of referenced and controlled resources which can be added to Verily Workbench resources. Provide widgets for adding each type of referenced and controlled resource to the user's current workspace.

emmarogge committed:
Create notebook introducing workspace resources

Create notebook introducing the types of referenced and controlled resources which can be added to Verily Workbench resources. Provide widgets for adding each type of referenced and controlled resource to the user's current workspace.

emmarogge committed:
Update screencast to one using test account.

emmarogge committed:
Encapsulate widgets in classes

Encapsulate each widget in a class for improved readability.

emmarogge committed:
Use string formatting inline for description

Use inline Python string formatting to ensure each description is cast to a string.

emmarogge committed:
Clean up imports

Remove unnecessary import of json package.

emmarogge committed:
Create notebook introducing VWB groups (#30)

* Create notebook with examples of VWB groups

Create a notebook containing widgets to enable users to execute various VWB group management operations including:
- View groups in which the user is a member/admin
- View membership of groups in which user is an admin
- Create a new VWB group
- Delete a VWB group
- Add or remove a single user from VWB group
- Add or remove multiple users from VWB by providing a CSV file

* Refactor introduction to include ref to docs, CLI

* Update for usability & reviewer feedback

- De-duplicate widget input field names for each widget.
- Create hyperlinks in Overview section to corresponding section/cell.
- Use placeholder emails in template CSV file.
- Rephrase awkward add/remove content in overview.

* Clarify instructions for add/remove user widget

* Refactor to encapsulate widgets for readability

Refactored each widget type as a class in order to maintain readability while preventing issues arising from variables with the same name. Moved the widget labels (with instructions) above the other fields so it looks reasonable after output is appended.

* Rephrase awkward instruction sentence

wnojopra commented on first_hour_on_terra/workspace_resource_examples.ipynb:
```diff
@@ -0,0 +1,1403 @@
+   "source": [
+    "#### 1. Add a GitHub repository\n",
+    "\n",
+    "Both public and private GitHub repositories can be added to data collections. In order to add a private GitHub repository, you must first [set up your Terra SSH key](https://terra-docs.api.verily.com/docs/how_to_guides/terra_ssh_key_guide) and should provide a repo URL of the format `git@github.com` instead of `http://`. No additional setup is required to add a public GitHub repository.\n",
```
```suggestion
    "Both public and private GitHub repositories can be added to data collections. In order to add a private GitHub repository, you must first [set up your Terra SSH key](https://terra-docs.api.verily.com/docs/how_to_guides/terra_ssh_key_guide) and should provide a repo URL of the format `git@github.com` and not `http://`. No additional setup is required to add a public GitHub repository.\n",
```

emmarogge replied:
done!

wnojopra commented on first_hour_on_terra/workspace_resource_examples.ipynb:
```diff
@@ -0,0 +1,1403 @@
+   "source": [
+    "class AddGcsBucketWidget(object):\n",
+    "    def __init__(self):\n",
+    "        self.label = widgets.Label(value = 'Please provide appropriate values in the input boxes.')\n",
```
Just like with the data collections notebook, I think it needs to be more clear what the user is expected to input here. Please add instructions here that tells the user that they're expected to make up a repo name and description and give examples of good names. Also give an example of a github url (maybe a public one). I think the cloning options also need some explanations.

emmarogge replied:
I have now gone in and provided detailed explanations for each of the input parameters for each widget. It makes for a longer notebook, overall, but hopefully one that's more useful/less frustrating for users.

wnojopra commented on first_hour_on_terra/workspace_resource_examples.ipynb:
```diff
@@ -0,0 +1,1403 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 3. Create a GCP notebook\n",
```
Even though the command is called `create gcp-notebook`, I really think we should call it a cloud environment. https://terra-docs.api.verily.com/docs/how_to_guides/cloud_environments/#accessing-the-terra-command-line-tool-from-your-cloud-environment for example correctly calls it a cloud environment even though the command line example calls it a gcp-notebook.

In fact, I'm going to file a ticket to rename it in the CLI.

emmarogge replied:
awesome! I will update this accordingly once the CLI change lands.

emmarogge committed:
Merge remote-tracking branch 'origin/emmarogge/resources' into emmarogge/resources

emmarogge committed:
Swap "instead of" for "and not"

emmarogge committed:
Add section to explain cloning instructions

Add a section that explains the nuances of each possible cloning instruction value.

emmarogge committed:
Provide detailed explanation of widget input parameters

Provide more details about each widget's input parameters and provide examples of appropriate values. Link to Google Cloud documentation where appropriate.

emmarogge committed:
Clear cell outputs

emmarogge commented:
Closing this PR, continuing in #33
